File: server/models/history.js
const mongoose = require("mongoose");

const historySchema = mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
  },
  imageUrl: {
    type: String,
    required: true,
  },
  time: {
    type: Date,
    required: true,
  },
});

const History = mongoose.model("History", historySchema);

module.exports = History;

File: lib/models/history.dart
class History {
  final String name;
  final String imageUrl;
  final DateTime time;

  History({
    required this.name,
    required this.imageUrl,
    required this.time,
  });

  factory History.fromJson(Map<String, dynamic> json) {
    return History(
      name: json['name'],
      imageUrl: json['imageUrl'],
      time: DateTime.parse(json['time']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'imageUrl': imageUrl,
      'time': time.toIso8601String(),
    };
  }
}
